addons/website_sale/controllers/main.py

```diff
@@ -330,7 +330,7 @@ def cart(self, **post):
         cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
         order = request.website.sale_get_order()
         if order:
-            from_currency = pool.get('product.price.type')._get_field_currency(cr, uid, 'list_price', context)
```

Comment:
Everytime you get this in the code:
   pool.get('product.price.type')._get_field_currency(cr, uid, 'list_price', context)
You must replace by getting the currency of the company of the current user.